Core/PARStore.h

```diff
@@ -95,17 +96,41 @@ extern NSString *PARStoreDidSyncNotification;
 // This method returns an array of PARChange instances. It should not be called from within a transaction, or it will fail.
 - (NSArray *)fetchChangesSinceTimestamp:(nullable NSNumber *)timestamp;
 
+/// This method returns an array of PARChange instances for the device identifier passed in.
+/// It should not be called from within a transaction, or it will fail.
+/// Pass in nil for the device identifier to get results for all devices.
+- (NSArray *)fetchChangesSinceTimestamp:(nullable NSNumber *)timestampLimit forDeviceIdentifier:(nullable NSString *)deviceID;
+
+/// This method returns an array of PARChange instances for the device identifier passed in, between (including) the timestamps passed.
+/// It should not be called from within a transaction, or it will fail.
+/// Pass in nil for the device identifier to get results for all devices.
+/// Pass nil for either timestamp to have an open range.
+- (NSArray *)fetchChangesFromTimestamp:(nullable NSNumber *)firstTimestamp toTimestamp:(nullable NSNumber *)lastTimestamp forDeviceIdentifier:(nullable NSString *)deviceID;
+
 // TODO: error handling
 
 @end
 
 
+// Accesss for backends that need to import data from other devices via cloud services.
+@interface PARStore (RemoteUpdates)
+
+/// This method can only append changes after existing entries.
+/// In other words, only changes that have a timestamp greater than all
+/// existing entries will be inserted into the store.
+- (void)appendChanges:(NSArray *)changes forDeviceIdentifier:(NSString *)deviceIdentifier;
+
+@end
+
+
 @interface PARChange : NSObject
 + (PARChange *)changeWithTimestamp:(NSNumber *)timestamp parentTimestamp:(nullable NSNumber *)parentTimestamp key:(NSString *)key propertyList:(id)propertyList;
++ (PARChange *)changeWithPropertyListRepresentation:(id)propertyListRepresentation;
```

Comment:
`propertyDictionary` also allows for less confusion with the already existing `propertyList` property on `PARChange` as well btw